arborist/server_test.go

```diff
@@ -915,6 +921,49 @@ func TestServer(t *testing.T) {
 			assert.Equal(t, []string{"/a/b"}, result.Subresources, msg)
 		})
 
+		t.Run("Overwrite", func(t *testing.T) {
+			w := httptest.NewRecorder()
+			body := []byte(`{
+				"name": "Gödel",
+				"subresources": [
+					{
+						"name": "Escher",
+						"subresources": [{"name": "Bach"}]
+					}
+				]
+			}`)
+			req := newRequest("PUT", "/resource", bytes.NewBuffer(body))
+			handler.ServeHTTP(w, req)
+			if w.Code != http.StatusCreated {
+				httpError(t, w, "couldn't create resource using PUT")
+			}
+			expected := struct {
+				_ interface{} `json:"created"`
+			}{}
+			err = json.Unmarshal(w.Body.Bytes(), &expected)
+			if err != nil {
+				httpError(t, w, "couldn't read response from resource creation")
+			}
+			bachTag := getTagForResource("Gödel.Escher.Bach")
+			// now PUT over the same resource, but keep the subresources
+			w = httptest.NewRecorder()
+			body = []byte(`{
+				"name": "Godel",
+				"subresources": [
+					{"name": "Escher", "subresources": [{"name": "Bach"}]},
+					{"name": "completeness_theorem"}
+				]
+			}`)
+			req = newRequest("PUT", "/resource", bytes.NewBuffer(body))
+			handler.ServeHTTP(w, req)
+			if w.Code != http.StatusCreated {
+				httpError(t, w, "couldn't update resource using PUT")
+			}
+			newBachTag := getTagForResource("Godel.Escher.Bach")
+			assert.Equal(t, bachTag, newBachTag, "subresource tag changed after PUT")
```

Comment:
shouldn't the existing resource tag have stayed the same after PUT? I thought the idea was that PUT will not change the tags of subresources that already exist, create ones that don't already exist, and remove subresources no longer in the tree?

Reply:
I misunderstood :man_facepalming: nevermind